mm/umm_heap/umm_realloc.c

```diff
@@ -83,7 +77,7 @@ FAR void *realloc(FAR void *oldmem, size_t size)
       mem = mm_realloc(USR_HEAP, oldmem, size);
       if (!mem)
         {
-          brkaddr = sbrk(size);
+          brkaddr = sbrk(size < 1 ? 1 : size);
```

Comment:
The actual size needed is actually larger than one when size is zero:  It is the minimum chunk size.

Reply:
```
TLSF_API void* tlsf_realloc(tlsf_t tlsf, void* ptr, size_t size)
{
	control_t* control = tlsf_cast(control_t*, tlsf);
	void* p = 0;

	/* Zero-size requests are treated as free. */
	if (ptr && size == 0)
	{
		tlsf_free(tlsf, ptr);
	}
```
so `tlsf_realloc` is not compliant with the latest POSIX spec

Reply:
Will handle according to https://pubs.opengroup.org/onlinepubs/7908799/xsh/brk.html